src/libcore/hash/sip.rs

```diff
@@ -193,6 +193,14 @@ impl Hasher for SipHasher {
     }
 
     #[inline]
+    fn delimit(&mut self, len: usize) {
+        // skip the first delimiter
+        if self.length > 0 {
+            self.write_usize(len);
```

Comment:
Or wait, is that a problem? It's just two elements with the same hash, not arbitrary many with the same hash. So the example it's not enough for a hash flooding attack.